ui/src/components/FileInputComponent/FileInputComponent.test.tsx

```diff
@@ -375,5 +376,7 @@ test('Check FileInputComponent encodes base64 correctly and passed it to handler
     await userEvent.upload(fileInput, testfile);
 
     // Check that handleChange is called with valid args.
-    expect(handleChange).toHaveBeenCalledWith(field, btoa(fileContent));
+    await act(async () => {
+        expect(handleChange).toHaveBeenCalledWith(field, btoa(fileContent));
```

Comment:
I am not sure about it. I am looking at the `act` docs:

> Wrap any code rendering and triggering updates to your components into act() calls.

Why does `expect` trigger update?

Reply:
ok i think i misunderstood the method too, seems like it is 'waiting' for all changes to happen, so maybe we should use it for  line with `await userEvent.upload(fileInput, testfile);`, but at the same time we are doing it already with await, dont we?

according to docs:
[act](https://legacy.reactjs.org/docs/testing-recipes.html#act)
When writing UI tests, tasks like rendering, user events, or data fetching can be considered as “units” of interaction with a user interface. react-dom/test-utils provides a helper called act() that makes sure all updates related to these “units” have been processed and applied to the DOM before you make any assertions:

Reply:
act() is necessary to ensure that all asynchronous operations related to the file upload are complete and reflected in the DOM before making assertions.

Reply:
using await with userEvent.upload() ensures that the file upload completes, but it doesn’t necessarily guarantee that all related state updates have been applied to the DOM before proceeding, so I believe we can add the act() in it too, wdyt?

Reply:
Probably the thing that Viktor tried to point out is that act ensures that all actions passed to this method are complete and reflected, but in code there is only
``` 
expect(handleChange).toHaveBeenCalledWith(field, btoa(fileContent));
```
passed to act method, so this code inside wont influence file component at all

mine point was that we might try to wrap  `    await userEvent.upload(fileInput, testfile);` inside act so sth like this 

```
    await act(async () => {
            await userEvent.upload(fileInput, testfile);
    });
    expect(handleChange).toHaveBeenCalledWith(field, btoa(fileContent));
```

But general question probably should be, if you managed to reproduce that error consecutively before the fix and it was working after it.

Reply:
I have tried with the many approaches to reproduce the error but not able to find any, also this thing is not breaking in any of the pipeline for various PR.